persistence-cassandra/javadsl/src/main/scala/com/lightbend/lagom/internal/javadsl/persistence/cassandra/CassandraReadSideHandler.scala

```diff
@@ -28,44 +29,58 @@ import scala.concurrent.{ ExecutionContext, Future }
  * Internal API
  */
 private[cassandra] abstract class CassandraReadSideHandler[Event <: AggregateEvent[Event], Handler](
-  session: CassandraSession, handlers: Map[Class[_ <: Event], Handler], dispatcher: String
+  session:    CassandraSession,
+  handlers:   Map[Class[_ <: Event], Handler],
+  dispatcher: String
 )(implicit ec: ExecutionContext) extends ReadSideHandler[Event] {
 
   private val log = LoggerFactory.getLogger(this.getClass)
 
   protected def invoke(handler: Handler, event: Event, offset: Offset): CompletionStage[JList[BoundStatement]]
 
   override def handle(): Flow[Pair[Event, Offset], Done, _] = {
-    akka.stream.scaladsl.Flow[Pair[Event, Offset]].mapAsync(parallelism = 1) { pair =>
-      handlers.get(pair.first.getClass.asInstanceOf[Class[Event]]) match {
-        case Some(handler) =>
-          for {
-            statements <- invoke(handler, pair.first, pair.second).toScala
-            done <- statements.size match {
-              case 0 => Future.successful(Done.getInstance())
-              case 1 => session.executeWrite(statements.get(0)).toScala
-              case _ =>
-                val batch = new BatchStatement
-                val iter = statements.iterator()
-                while (iter.hasNext)
-                  batch.add(iter.next)
-                session.executeWriteBatch(batch).toScala
-            }
-          } yield done
-        case None =>
-          if (log.isDebugEnabled)
-            log.debug("Unhandled event [{}]", pair.first.getClass.getName)
-          Future.successful(Done.getInstance())
+
+    def executeStatements(statements: JList[BoundStatement]): Future[Done] =
+      statements.size match {
+        case 0 => Future.successful(Done.getInstance())
+        case 1 => session.executeWrite(statements.get(0)).toScala
+        case _ =>
+          val batch = new BatchStatement
+          val iter = statements.iterator()
+          while (iter.hasNext)
+            batch.add(iter.next)
```

Comment:
`BatchStatement` has an `addAll(Iterable)` method... not sure why this wasn't using it in the first place.